lib/cli-engine.js

```diff
@@ -573,7 +572,9 @@ CLIEngine.prototype = {
             if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
                 executeOnFile(pattern);
             } else {
-                glob.sync(pattern, globOptions).forEach(executeOnFile);
+                glob.sync(pattern, globOptions).filter(function(filename) {
+                    return !ignoredPaths.contains(filename);
```

Comment:
Looks like we should put the ignore check in `executeOnFile` at this point.

Also this may have a bad impact on performance. That's why I was trying to use the `ignore` option for `glob`. Is it completely broken?